lib/db_ref.js

```diff
@@ -1,44 +1,47 @@
 'use strict';
 /**
  * A class representation of the BSON DBRef type.
- *
- * @class
- * @param {string} collection the collection name.
- * @param {ObjectID} oid the reference ObjectID.
- * @param {string} [db] optional db name, if omitted the reference is local to the current db.
- * @return {DBRef}
  */
-function DBRef(collection, oid, db, fields) {
-  if (!(this instanceof DBRef)) return new DBRef(collection, oid, db, fields);
+class DBRef {
+  /**
+   * Create a DBRef type
+   *
+   * @param {string} collection the collection name.
+   * @param {ObjectID} oid the reference ObjectID.
+   * @param {string} [db] optional db name, if omitted the reference is local to the current db.
+   * @return {DBRef}
+   */
+  constructor(collection, oid, db, fields) {
+    // check if namespace has been provided
+    const parts = collection.split('.');
+    if (parts.length === 2) {
+      db = parts.shift();
+      collection = parts.shift();
+    }
 
-  // check if namespace has been provided
-  var parts = collection.split('.');
-  if (parts.length === 2) {
-    db = parts.shift();
-    collection = parts.shift();
+    this._bsontype = 'DBRef';
+    this.collection = collection;
+    this.oid = oid;
+    this.db = db;
+    this.fields = fields || {};
   }
 
-  this._bsontype = 'DBRef';
-  this.collection = collection;
-  this.oid = oid;
-  this.db = db;
-  this.fields = fields || {};
-}
-
-/**
- * @ignore
- * @api private
- */
-DBRef.prototype.toJSON = function() {
-  var o = {
-    $ref: this.collection,
-    $id: this.oid
-  };
+  /**
+   * @ignore
+   * @api private
+   */
+  toJSON() {
+    const o = Object.assign(
```

Comment:
This changes the order of assignment such that `this.db` comes after `this.fields`. Is that expected?

Reply:
It's different, but I'm having trouble figuring out when you would want to do this or why order of assignment matters? Do you have something in mind?

Reply:
I honestly don't expect it to be any different, I just wanted to note that it is. Hypothetically, someone could have `this.fields.$db` that they expect to overrride the value of `this.db`. But that is unlikely.